### PR TITLE
Update some deps

### PR DESCRIPTION
**Pull request by nrc**
To reduce the number of duplicated deps in TiKV

PTAL @Hoverbear @hicqu

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ prost-codec = ["raft-proto/prost-codec"]
 
 # Make sure to synchronize updates with Harness.
 [dependencies]
-log = ">0.2"
+log = "0.4"
```
Comment (BusyJay) [inline Cargo.toml]: I think `>0.2` includes `0.4`?

Comment (Hoverbear) [reply Cargo.toml]: Perhaps it is necessary for `slog-stdlog`?

File: src/lib.rs [hunk 2]
```diff
@@ -374,6 +374,8 @@ before taking old, removed peers offline.
 #![recursion_limit = "128"]
 // This is necessary to support prost and rust-protobuf at the same time.
 #![allow(clippy::identity_conversion)]
+// This lint recommends some bad choices sometimes.
```
Comment (Hoverbear) [inline src/lib.rs]: Is there an issue to reference here?

Comment (nrc) [reply src/lib.rs]: split off into https://github.com/pingcap/raft-rs/pull/282

**Comment by nrc** [comment]
PTAL @hicqu @brson (there are no material changes, just split out some of it into https://github.com/pingcap/raft-rs/pull/282)

File: src/lib.rs (hunk 2 above)
Comment (Hoverbear) [reply src/lib.rs]: They're still part of this PR so #282 blocks this :(

Comment (nrc) [reply src/lib.rs]: I put them in a separate commit so it can still be reviewed. Just needs a rebase now.

**Pull request merged**